jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/AbstractQueryRewriter.java

```diff
@@ -460,4 +460,25 @@ public Object getResultSetValue(ResultSet resultSet, int columnIndex, Column col
         }
         return resultSet.getObject(columnIndex);
     }
+
+    protected boolean isSupportedVersion(String databaseProductName, int databaseVersion) {
+
+        if(databaseProductName.equals(_dataContext.getDatabaseProductName())
+                && databaseVersion <= databaseVersionToInt(_dataContext.getDatabaseVersion())) {
+            return true;
+        }
+        return false;
+    }
+
+    private int databaseVersionToInt(String version) {
+        int firstDot = -1;
+        if(version != null) {
+            firstDot = version.indexOf('.');
+        }
+        if(firstDot >= 0) {
+            return Integer.valueOf(version.substring(0, firstDot));
```

Comment:
This could throw an exception if the version from the DB is something like `R12.0`.